modules/post/windows/gather/get_bookmarks.rb

```diff
@@ -0,0 +1,68 @@
+class MetasploitModule < Msf::Post
+    include Msf::Post::File
+    include Msf::Post::Windows::UserProfiles
+    def initialize(info = {})
+      super(
+        update_info(
+          info,
+          'Name' => 'Bookmarked Sites Retriever',
+          'Description' => %q{
+            This module discovers information about a target by retrieving their bookmarked websites on Google Chrome, Opera and Microsoft Edge.
+          },
+          'License' => MSF_LICENSE,
+          'Author' => [ 'jerrelgordon'],
+          'Platform' => [ 'win' ],
+          'SessionTypes' => [ 'meterpreter', 'shell' ]
+        )
+   )
+    end
+
+    def run
+      get_bookmarks("GoogleChrome") # gets bookmarks for google chrome
+      get_bookmarks("Opera") # gets bookmarks for opera
+      get_bookmarks("Edge") # gets bookmarks for edge
```

Comment:
Is there a way that this module can search for the existence of each browser first instead of performing the same loop for each browser listed here?

Reply:
Great suggestion! I will give the approach some thought.